src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -500,7 +518,7 @@ void OnOffServer::initOnOffServer(chip::EndpointId endpoint)
 
         bool onOffValueForStartUp = false;
         Status status             = getOnOffValueForStartUp(endpoint, onOffValueForStartUp);
-        if (status == Status::Success)
+        if (status == Status::Success && GetBootReason() != BootReasonType::kSoftwareUpdateCompleted)
```

Comment:
So if StartUpOnOff is true but OnOff is false and we do an OTA restart, what happens with the OnMode/CurrentMode bits below, and is it the right behavior?

Separately, what happens if OTA happens and then 3 days later a new endpoint that has an On/Off cluster is added (which will call this code, I believe)?  That said, that handling might have been odd already, since arguably in that case doing StartUpOnOff and OnMode is weird.  So a followup issue on this is probably best.

Reply:
> So if StartUpOnOff is true but OnOff is false and we do an OTA restart, what happens with the OnMode/CurrentMode bits below, and is it the right behavior?

Good point! I've tried to minimize changes, but looks like it should be fixed too. I've updated the PR

> Separately, what happens if OTA happens and then 3 days later a new endpoint that has an On/Off cluster is added (which will call this code, I believe)? That said, that handling might have been odd already, since arguably in that case doing StartUpOnOff and OnMode is weird. So a followup issue on this is probably best.

For now the behaviour will be "leave as is". So onOff value will be set to its previous value. And because we have no previous value, it will be the default value.

Reply:
> For now the behaviour will be "leave as is". So onOff value will be set to its previous value.

No, it'll be set to the StartUpOnOff value, since the boot reason is "OTA".... That's the same as the behavior before your PR, though.  Again, this needs a followup issue.

Reply:
Looks like we are talking about different things. I agree that this needs a followup issue, I'll create it. Just lets clarify.

**Before this PR**:
In case of any boot reason:
- `newOnOff = f(oldOnOff, StartUpOnOff)`
- if `newOnOff == true`: `CurrentMode = OnMode`

**After this PR**:
In case of OTA boot reason:
- `newOnOff = oldOnOff` (thats what I call "leave as is")
- if `newOnOff == true`: `CurrentMode = OnMode`

In case of any other boot reason:
- `newOnOff = f(oldOnOff, StartUpOnOff)`
- if `newOnOff == true`: `CurrentMode = OnMode`

So after this PR the behaviour in case of scenario "if OTA happens and then 3 days later a new endpoint that has an On/Off cluster is added" will be changed from `newOnOff = f(oldOnOff, StartUpOnOff)` to `newOnOff = oldOnOff` which in such case will be `newOnOff = defaultOnOff`.
The problem is that the spec doesn't  specify such behaviour. All we know is "StartUpOnOff Attribute
defines the desired startup behavior of a device when it is **supplied with power**" and probably shouldn't affect the behaviour of adding new endpoint to powered device.

Also could you please give an example of the device where such case is possible?

Reply:
> Also could you please give an example of the device where such case is possible?

A bridge, for starters.  But there are other possibilities too (e.g. a device that can have new hardware added to it when it's powered on).

Reply:
> So a followup issue on this is probably best

https://github.com/project-chip/connectedhomeip/issues/34452